KIPs/kip-81.md

```diff
@@ -0,0 +1,335 @@
+
+The Voting contract revolves around proposals. A proposal describes the changes to be made on the Klaytn blockchain. A proposal not only contains textual descriptions of the change, but it may also include specifications of several on-chain transactions, or actions. The actions are executed on behalf of a voting contract after the proposal passes. The actions allow governance voting to enforce changes on the blockchain.
+
+```solidity
+abstract contract Voting {
+  struct Proposal {
+    string description;  // human readable description
+    address[] targets;   // 0 or more transactions
+    uint256[] values;
+    bytes[] calldatas;
+  }
+}
+```
+
+A proposer can submit governance proposals by calling `propose()` function. The voting timeline starts with the submission of proposals to the voting contract. A proposer can also cancel a proposal she has submitted, if the proposal had not been executed yet.
+
+A voter casts votes on behalf of one of the GCs. The vote choice is one of yes, no, or abstain.
+
+An executor triggers the execution of attached transactions in passed proposals. The execution is subdivided into two steps. First, the executor queues the transactions. After a set amount of delay, the executor can send the transactions. Execution delay here gives enough time for the community to recognize the upcoming change and perform a final check about the transaction.
+
+There will be a secretariat account stored in the Voting contract. The account is initially managed by the Klaytn Foundation, and later it can be replaced by governance voting. The secretariat has the proposer and executor role.
+
+```solidity
+abstract contract Voting {
+
+  function propose(
+    string description, address[] targets, uint256[] values, bytes[] calldatas)
+    virtual returns (uint256 proposalId);
+  function cancel(uint256 proposalId) virtual;
+
+  enum VoteChoice{ Unknown, Yes, No, Abstain };
+  function castVote(uint256 proposalId, uint32 choice) virtual;
+
+  function queue(uint256 proposalId) virtual onlyExecutor;
+  function execute(uint256 proposalId) virtual onlyExecutor;
+}
+```
+
+
+Following timing parameters influence the Voting contract.
+
+| Name              | Meaning                                                                                   | Value     |
+|----------------   |----------------------------------------------------------------------------------------   |--------   |
+| `votingDelay`     | Delay from proposal submission to voting start. Equals to the length of Pending state.    | 7 days    |
+| `votingPeriod`    | Duration of the voting. Equals the length of Active state.                                | 7 days    |
+| `queueTimeout`    | Grace period given to executors for queue() passed proposals.                             | 7 days    |
+| `execDelay`       | A minimum delay before a queued transaction can be executed.                              | 2 days    |
```

Comment:
execDelay is set per each transaction?
For example, if we have multiple transactions T1, T2, and T3 for proposal P1 and transactions T4 and T5 for proposal P2, how is the exacDelay set?